### PR TITLE
docs(CMF): user doc

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
No use doc

**What is the new behavior?**
CMF user doc ready for gh-pages

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

File: docs/_includes/disqus.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<div id="disqus_thread"></div>
```
Comment (talend-glorieux) [inline docs/_includes/disqus.html]: Guess that file can be removed.

File: docs/_includes/google_analytics.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<script>
+  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
+
+  ga('create', '{{ site.google_analytics_id }}', 'auto');
+  ga('send', 'pageview');
+</script>
```
Comment (talend-glorieux) [inline docs/_includes/google_analytics.html]: That file should be removed

File: docs/_layouts/default.html [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                <div id="disqus" class="col-sm-10">
+                    {% include disqus.html %}
+                </div>
+              {% else %}
+                <div id="disqus" class="col-sm-12">
+                    {% include disqus.html %}
+                </div>
+              {% endif %}
+            </div>
+            {% endif %}
```
Comment (talend-glorieux) [inline docs/_layouts/default.html]: No need for the disqus part.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

File: docs/_posts/2017-02-28-core-store.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+## Reducers
+
+The CMF store comes with some reducers out of the box
+* router reducer from [react-router-redux](https://github.com/reactjs/react-router-redux)
+* CMF internal reducer (settings, ...)
+
+
+## Middlewares
+
+The CMF store comes with some middleware out of the box
```
Comment (frassinier) [inline docs/_posts/2017-02-28-core-store.md]: middlewares

File: docs/_includes/disqus.html (hunk 1 above)
Comment (jsomsanith) [inline docs/_includes/disqus.html]: Done

File: docs/_includes/google_analytics.html (hunk 2 above)
Comment (jsomsanith) [inline docs/_includes/google_analytics.html]: Done

File: docs/_layouts/default.html (hunk 3 above)
Comment (jsomsanith) [inline docs/_layouts/default.html]: Done

File: docs/_posts/2017-02-28-how-to-add-a-new-page-.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
+---
+layout: page
+title: "How to add a new page ?"
+category: faq
+date: 2017-02-28 10:19:22
+order: 3
+---
+
+To add a new page on a new route using CMF, you need to perform the steps :
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-add-a-new-page-.md]: these steps

File: docs/_posts/2017-02-28-how-to-add-a-new-page-.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+            role={this.props.link ? 'link' : null}
+        >
+            {this.props.label}
+        </Button>
+    );
+}
+MyComponent.propTypes = {
+    label: PropTypes.string.isRequired,
+    link: PropTypes.bool,
+    onClick: PropTypes.func.isRequired, // the pure component wait for the dispatch action
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-add-a-new-page-.md]: waits

File: docs/_posts/2017-02-28-how-to-connect-a-container-to-cmf-.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+* the register
+* the action utility
+* ...
+
+CMF provides a [React Context](https://facebook.github.io/react/docs/context.html) system.
+
+## App
+
+CMF provides an `<App />` component to bootstrap pour app. This component
+* plug [react-redux](https://github.com/reactjs/react-redux) `Provider` with the store you pass
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-connect-a-container-to-cmf-.md]: plugs

File: docs/_posts/2017-02-28-how-to-connect-a-container-to-cmf-.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+* plug CMF `RegistryProvider`
+* connect [react-router](https://github.com/ReactTraining/react-router) with the `history you pass and the `routes` [settings]({{ site.baseurl }}{% link _posts/2017-02-28-core-settings.md %})
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-connect-a-container-to-cmf-.md]: connects

File: docs/_posts/2017-02-28-how-to-connect-a-container-to-cmf-.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+//init your registry, settings, store
+...
+
+render(
+	<App store={store} history={syncHistoryWithStore(browserHistory, store)} />,
+	document.getElementById('app')
+);
+```
+
+The `<App />` component wrap your app with `react-redux` provider and CMF `RegistryProvider`.
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-connect-a-container-to-cmf-.md]: wraps

File: docs/_posts/2017-02-28-how-to-manage-collections-.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
+---
+layout: page
+title: "How To Manage Collections ?"
+category: faq
+date: 2017-02-28 10:18:56
+order: 1
+---
+
+CMF has a system to manage collections by a combination of store middleware, action, reducer.
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-manage-collections-.md]: of middleware, action and reducer store

File: docs/_posts/2017-02-28-how-to-manage-collections-.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+By using those entities, you will have the collections in your app state `state.cmf.collections`. This object is a dictionary that store a collection under unique identifiers.
```
Comment (frassinier) [inline docs/_posts/2017-02-28-how-to-manage-collections-.md]: stores

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
+layout: page
+title: "What is CMF ?"
+category: home
+date: 2017-02-28 10:09:02
+order: 1
+---
+
+CMF stands for Content Management Framework. It helps you build an application in React with a system to configure your contents and much more.
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: to build

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+It combines
+* [Redux](http://redux.js.org/) to manage app state
+* [React Router](https://github.com/ReactTraining/react-router) to manage your routes
+
+Based on a `component / action creator` registry and a `settings` format, you configure your entire app
+* the routes
+* what component to load on a route
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: which

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+## How CMF works
+
+*Registry*
+* Register your `components` with unique names.
+* Register your `action creators` with unique names.
+
+*Settings*
+* Settings > routes map `urls` to registered `components` names and `views` settings names.
+* Settings > views contains the `components` props and can refers to `actions` settings names.
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: refer

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+* Settings > actions contains either static actions definition or can refers to registered `action creators` names.
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: refer

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+CMF stands for Content Management Framework. It helps you build an application in React with a system to configure your contents and much more.
+
+It combines
+* [Redux](http://redux.js.org/) to manage app state
+* [React Router](https://github.com/ReactTraining/react-router) to manage your routes
+
+Based on a `component / action creator` registry and a `settings` format, you configure your entire app
+* the routes
+* what component to load on a route
+* what props to pass to this component to load
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: which

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+* what action is dispatched
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: which

File: docs/_posts/2017-02-28-what-is-cmf.md [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+* Register your `action creators` with unique names.
+
+*Settings*
+* Settings > routes map `urls` to registered `components` names and `views` settings names.
+* Settings > views contains the `components` props and can refers to `actions` settings names.
+* Settings > actions contains either static actions definition or can refers to registered `action creators` names.
+
+*Bootstrap and runtime*
+* Bootstrap your app with the provided `<App />` component
+* CMF instantiate the right component on a requested route, injecting the props
```
Comment (frassinier) [inline docs/_posts/2017-02-28-what-is-cmf.md]: instantiates

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://222.talend.surge.sh)

**Pull request merged**